### PR TITLE
Add theBeat

### PR DESCRIPTION
**Pull request by vicr123**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions.
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.vicr123.thebeat.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+app-id: com.vicr123.thebeat
+runtime: org.kde.Platform
+runtime-version: '5.15'
```
Comment (hfiguiere) [inline com.vicr123.thebeat.yml:3]:
```suggestion
runtime-version: '5.15-21.08'
```

File: com.vicr123.thebeat.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  - '/bin/cd*'
+  - '/bin/iso*'
+  - '/bin/mmc-tool'
+  - '/bin/taglib-config'
+  - '/bin/tltrdesktop'
+  - '/lib/debug'
+  - '/lib/pkgconfig'
+modules:
+  - name: taglib
+    buildsystem: cmake
```
Comment (hfiguiere) [inline com.vicr123.thebeat.yml:28]:
```suggestion
    buildsystem: cmake-ninja
```

File: com.vicr123.thebeat.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      - THELIBS_INSTALL_QT_PLUGINS = /app/lib/qt/plugins
+      - '-after'
+      - QT -= thelib
+      - QT += dbus
+      - LIBS += -L/app/lib -lthe-libs
+      - INCLUDEPATH += /app/include/the-libs
+    sources:
+      - type: git
+        url: https://github.com/vicr123/contemporary-theme
+        branch: master
```
Comment (hfiguiere) [inline com.vicr123.thebeat.yml:66]: you need a specific revision / tag. not a branch.

File: com.vicr123.thebeat.yml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      - THELIBS_INSTALL_SETTINGS = /app/etc
+      - '-after'
+      - QT -= thelib
+      - QT += dbus
+      - LIBS += -L/app/lib -lthe-libs
+      - INCLUDEPATH += /app/include/the-libs
+    sources:
+      - type: git
+        tag: "v3.1.1"
+        commit: "9bb7b3eaa8ff554009d60b807cd45b8371740cf1"
```
Comment (hfiguiere) [inline com.vicr123.thebeat.yml:90]: there is no URL

File: com.vicr123.thebeat.yml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        branch: master
+  - name: contemporary-icons
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/share/icons/contemporary
+      - cp -r * /app/share/icons/contemporary/
+    sources:
+      - type: git
+        tag: "v1.10"
+        commit: "97be306b78006d248f692a480ae5eaa48eead607"
```
Comment (hfiguiere) [inline com.vicr123.thebeat.yml:75]: you are missing the URL to the repository

File: com.vicr123.thebeat.yml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    sources:
+      - type: archive
+        url: https://github.com/rocky/libcdio-paranoia/archive/release-10.2+2.0.1.tar.gz
+        sha256: 7a4e257c85f3f84129cca55cd097c397364c7a6f79b9701bbc593b13bd59eb95
+  - name: the-libs
+    buildsystem: qmake
+    sources:
+      - type: git
+        tag: "v5.1"
+        commit: "d185cc188bd105e1436605fa2f9f0d5c74591fbe"
```
Comment (hfiguiere) [inline com.vicr123.thebeat.yml:52]: you are missing the URL to the repository

**Review by hfiguiere — CHANGES_REQUESTED** [review]
it can't even build right now:

**Comment by vicr123** [comment]
My apologies, that should all be resolved now.

**Comment by hfiguiere** [comment]
bot, build com.vicr123.thebeat

**Comment by hfiguiere** [comment]
appdata fail to validate:
```
builddir/files/share/appdata/com.vicr123.thebeat.appdata.xml: FAILED:
• tag-missing           : <release> required
• tag-missing           : <url> is not present
```

**Comment by barthalion** [comment]
bot, build com.vicr123.thebeat

File: com.vicr123.thebeat.yml [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+runtime-version: '5.15-21.08'
+sdk: org.kde.Sdk
+command: thebeat
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --share=network
+  - --own-name=org.mpris.MediaPlayer2.theBeat
+  - --filesystem=home
```
Comment (barthalion) [inline com.vicr123.thebeat.yml:12]: Any reason this is not xdg-music?

Comment (vicr123) [reply com.vicr123.thebeat.yml:12]: There's an add to library function which simply tells the app to search for music in a different folder. Are permissions for that folder preserved between each startup?

Comment (barthalion) [reply com.vicr123.thebeat.yml:12]: To be honest, I'm not certain. If I execute `flatpak permission-list | grep -E 'read|write`, I see a number of files that I would assume application still have access to, but I don't know how uniform directory support across distros is. Let's keep it as is.

**Pull request closed without merging**